lib/resque/worker.rb

```diff
@@ -269,11 +234,29 @@ def work(interval = 5.0, &block)
 
       unregister_worker
     rescue Exception => exception
-      unless exception.class == SystemExit && !@child && run_at_exit_hooks
-        log_with_severity :error, "Failed to start worker : #{exception.inspect}"
+      return if exception.class == SystemExit && !@child && run_at_exit_hooks
+      log_with_severity :error, "Failed to start worker : #{exception.inspect}"
+      unregister_worker(exception)
+    end
+
+    def work_one_job(job = nil, &block)
```

Comment:
Do you need a test for this public interface that it's now taking an argument for a job?